src/js/common/SearchParams.js

```diff
@@ -0,0 +1,159 @@
+"use strict";
+
+define([], () => {
+  /** IDs used to encode search parameters to save map location to URL.  */
+  const saveToUrlDestinationIds = {
+    LATITUDE: "lt",
+    LONGITUDE: "ln",
+    HEIGHT: "ht",
+    HEADING: "hd",
+    PITCH: "p",
+    ROLL: "r",
+  };
+
+  /**
+   * A map from search parameter key to the actual keys used by the application.
+   */
+  const paramIdToDestinationKey = {
+    [saveToUrlDestinationIds.LATITUDE]: "latitude",
+    [saveToUrlDestinationIds.LONGITUDE]: "longitude",
+    [saveToUrlDestinationIds.HEIGHT]: "height",
+    [saveToUrlDestinationIds.HEADING]: "heading",
+    [saveToUrlDestinationIds.PITCH]: "pitch",
+    [saveToUrlDestinationIds.ROLL]: "roll",
+  };
+
+  /** Destination IDs plus enabled layers. */
+  const saveToUrlIds = {
+    ...saveToUrlDestinationIds,
+    ENABLED_LAYERS: "el",
+  };
```

Comment:
Done, though I won't call it allParamIds because my intention is that this searchparams library can be used in the future for different search params, not just ones related to saving the view to the URL. I think they belong all in one place because they must be unique and must not be reused for different purposes by different features.